### PR TITLE
Remove "globbing" for Math

### PR DESCRIPTION
**Pull request by oshadura**
To be done for this PR:
- Update math/unuran directory

and for next iteration try to move components to use only ROOT_STANDARD_LIBRARY_PACKAGE():
genvector & smatrix.

### CODE REVIEW

File: math/smatrix/CMakeLists.txt [hunk 1]
```diff
@@ -4,18 +4,45 @@
 
 set(libname Smatrix)
 
-ROOT_GENERATE_DICTIONARY(G__${libname}    Math/SMatrix.h     Math/SVector.h
-                                          Math/SMatrixDfwd.h Math/SMatrixFfwd.h
-                                          MODULE ${libname}
-                                          LINKDEF LinkDef.h OPTIONS "-writeEmptyRootPCM"
-                                          DEPENDENCIES Core MathCore)
+ROOT_GENERATE_DICTIONARY(G__${libname}
```
Comment (amadio) [inline math/smatrix/CMakeLists.txt]: I usually drop the `libname` variable when updating files like this. Feel free to drop it here too and use `Smatrix` directly.

File: math/vecops/CMakeLists.txt [hunk 2]
```diff
@@ -2,26 +2,19 @@
...
-set(SOURCES
-  src/RAdoptAllocator.cxx
-  src/RVec.cxx
-)
-
-if(vdt)
-  # needed for rootcling, to be removed
-  include_directories(${VDT_INCLUDE_DIRS})
-endif()
+include_directories(${VDT_INCLUDE_DIRS})
```
Comment (amadio) [inline math/vecops/CMakeLists.txt]: Could you please keep this part as before (i.e., inside the `if`)?

**Comment by oshadura** [comment]
Windows failure is not connected to my changes..

File: math/fftw/CMakeLists.txt [hunk 3]
```diff
@@ -3,9 +3,23 @@
...
+  HEADERS
+    TFFTComplex.h
+    TFFTComplexReal.h
+    TFFTRealComplex.h
+    TFFTReal.h
+  SOURCES
+    src/TFFTComplex.cxx
+    src/TFFTComplexReal.cxx
+    src/TFFTRealComplex.cxx
+    src/TFFTReal.cxx
```
[the remark this answers is not in the document]
Comment (oshadura) [reply math/fftw/CMakeLists.txt]: Done!

File: math/mathmore/CMakeLists.txt [hunk 4]
```diff
@@ -2,32 +2,83 @@
...
+    Math/LinkDef.h
+  LIBRARIES
+    ${GSL_LIBRARIES}
+  DEPENDENCIES
+    MathCore
+  BUILTINS
+    GSL
+)
 
+target_include_directories(MathMore PUBLIC ${GSL_INCLUDE_DIR})
```
[the remark this answers is not in the document]
Comment (oshadura) [reply math/mathmore/CMakeLists.txt]: Added target_link_libraries and I tested with builtin_gsl, seems to work for me.
If you want I can move to old style.

[the remark this answers is not in the document]
Comment (amadio) [reply math/mathmore/CMakeLists.txt]: I just meant that we need to keep an eye for failures due to builtins, but if it works, it's ok.

**Comment by oshadura** [comment]
@phsft-bot  build!

**Comment by oshadura** [comment]
@lmoneta, can I ask you please to check this PR? Thank you!

**Comment by lmoneta** [comment]
Hi Oxana, 

The configuration files in  mathcore/build and mathmore/build genvector/build ands matrix/build are used for stand-alone build of the packages. For just downloading them and building with simple ./configure/make.
I think for the moment it is better to keep them, but in case we remove them, I would not put these commit in this PR but open a separated one for this.

Thank you

Lorenzo

File: math/mathmore/CMakeLists.txt [hunk 5]
```diff
@@ -2,32 +2,84 @@
...
+    Math/GSLNLSMinimizer.h
+    Math/GSLSimAnMinimizer.h
+    Math/GSLMultiRootFinder.h
+    Math/GSLRandomFunctions.h
+    Math/Vavilov.h
+    Math/VavilovAccurate.h
+    Math/VavilovAccuratePdf.h
+    Math/VavilovAccurateCdf.h
+    Math/VavilovAccurateQuantile.h
+    Math/VavilovFast.h
```
Comment (amadio) [inline math/mathmore/CMakeLists.txt]: Could you please reorder alphabetically?

Comment (oshadura) [reply math/mathmore/CMakeLists.txt]: Done!

**Comment by oshadura** [comment]
@lmoneta and @amadio I think I resolved all your comments.
Lorenzo, I created a separate PR https://github.com/root-project/root/pull/2808 as you recommended in your review.

**Pull request merged**